src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/DiagnosticItem.cs

```diff
@@ -154,15 +148,34 @@ private void UpdateRuleSetFile(string pathToRuleSet, ReportDiagnostic value)
             var analyzerID = _analyzerItem.AnalyzerReference.Display;
             var rules = FindOrCreateRulesElement(ruleSetDocument, analyzerID);
             var rule = FindOrCreateRuleElement(rules, _descriptor.Id);
-            rule.Attribute("Action").Value = newAction;
+
+            if (value == ReportDiagnostic.Default)
+            {
+                // If the new severity is 'Default' we just delete the entry for the rule from the ruleset file.
+                // In the absence of an explicit entry in the ruleset file, the rule reverts back to its 'Default' severity.
```

Comment:
That isn't quite true. A diagnostic has a default severity, but if the rule set file lacks an entry the severity may still come from a project setting, or from a base rule set included by the current rule set.